clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -581,9 +577,7 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 // TODO i#5912: since we have the branch decoding now, we can handle
                 // this case.
                 type_is_instr_branch(
-                    shard->last_signal_context_.pre_signal_instr.instr.type) ||
-                shard->last_signal_context_.pre_signal_instr.instr.type ==
-                    TRACE_TYPE_INSTR_SYSENTER;
```

Comment:
Why was this removed?

Reply:
But both conditions have to be true, so removing this is not correct and will cause failures on 32-bit -- is that why the 32-bit test failed?  Did you investigate that failure?